File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.ts
function getMax (a:number, b: number): number{
    if (b>a) return b;
    return a
}

function maxProfit(prices: number[]): number {
    let left = 0, right = 1;
    let maxProfit = 0;
    
    while(right<prices.length){
        if(prices[left] < prices[right]){
            maxProfit = getMax(maxProfit, prices[right] - prices[left])
        }
        else {
            left = right
        }
        right = right + 1
    }
    return maxProfit
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.go
func getMax(a int, b int) int {
    if(b>a){
        return b
    }
    return a
}
func maxProfit(prices []int) int {
    left := 0
    right := 1
    maxProfit := 0
    
    for (right < len(prices)){
        if(prices[right] > prices[left]){
            maxProfit = getMax(maxProfit, prices[right]-prices[left])
        } else {
            left = right
        }
        
        right ++
    }
    return maxProfit
    
}